metaflow/plugins/events_decorator.py

```diff
@@ -597,9 +600,41 @@ def format_deploytime_value(self):
             # Entire trigger is a function (returns either string or dict)
             old_trig = trigger
             if isinstance(trigger, DeployTimeField):
+                # convert the trigger to string or dict
                 trigger = deploy_time_eval(trigger)
+                if is_stringish(trigger):
+                    pass
+                elif isinstance(trigger, dict):
+                    if "name" not in trigger:
```

Comment:
There is an interesting *fix* that this PR introduces. Previously this incorrect usage was also working for whatever reason:

```python
def flow_name_dict_func(ctx):
    # Is 'flow' correct??? docstring says 'name'
    return {"flow": "DeployTimeTriggerParams"}


@trigger_on_finish(flow=flow_name_dict_func)
class DeployTimeTriggerOnFinishFlow4(FlowSpec):
```

With the changes in this PR, Metaflow correctly raises an exception on the missing `name` in the dict

Reply:
There is an issue with the current PR though when trying out with the correct syntax.

```python
def flow_name_dict_func(ctx):
    # Is 'flow' correct??? docstring says 'name'
    return {"name": "DeployTimeTriggerParams"}
    # return {"name": "DeployTimeTriggerParams", "project": "TEST", "project_branch": "test_branch"}


@trigger_on_finish(flow=flow_name_dict_func)
class DeployTimeTriggerOnFinishFlow4(FlowSpec):
```
fails with `KeyError` on argo workflows due to `event["flow"]` not being set.